src/openforms/forms/api/serializers/form_variable.py

```diff
@@ -16,6 +24,64 @@
 from ...models import Form, FormDefinition, FormVariable
 
 
+@extend_schema_serializer(
+    examples=[
+        OpenApiExample(
+            name="Variable mapping example",
+            value={
+                "variable_key": "a_component_variable",
+                "target_path": ["path", "to.the", "target"],
+            },
+        )
+    ]
+)
+class ObjecttypeVariableMappingSerializer(serializers.Serializer):
+    """A mapping between a form variable key and the corresponding Objecttype attribute."""
+
+    variable_key = FormioVariableKeyField(
+        label=_("variable key"),
+        help_text=_(
+            "The 'dotted' path to a form variable key. The format should comply to how Formio handles nested component keys."
+        ),
+    )
+    target_path = serializers.ListField(
+        child=serializers.CharField(label=_("Segment of a JSON path")),
+        label=_("target path"),
+        help_text=_(
+            "Representation of the JSON target location as a list of string segments."
+        ),
+    )
+
+
+class FormVariableOptionsSerializer(JsonSchemaSerializerMixin, serializers.Serializer):
+    objects_api_group = PrimaryKeyRelatedAsChoicesField(
+        queryset=ObjectsAPIGroupConfig.objects.exclude(
+            Q(objects_service=None)
+            | Q(objecttypes_service=None)
+            | Q(drc_service=None)
+            | Q(catalogi_service=None)
+        ),
+        label=("Objects API group"),
+        required=False,
+        help_text=_("Which Objects API group to use."),
+    )
+    objecttype_uuid = serializers.UUIDField(
+        label=_("objecttype"),
+        required=False,
+        help_text=_("UUID of the objecttype in the Objecttypes API. "),
+    )
+    objecttype_version = serializers.IntegerField(
+        label=_("objecttype version"),
+        required=False,
+        help_text=_("Version of the objecttype in the Objecttypes API."),
+    )
+    variables_mapping = ObjecttypeVariableMappingSerializer(
+        label=_("variables mapping"),
+        many=True,
+        required=False,
+    )
+
```

Comment:
yeah so this is too naive :D while it achieves what we want in the ticket, it's really something that must be defined in the objects API prefill plugin itself. It's a very similar thing like the registrations plugin `configuration_options` serializer, which must be defined on the plugin itself and called dynamically during validation.